### DISCUSSION TITLE
Question about reference frame in MTEX

### DISCUSSION THREAD

**Discussion by A-Bourceret** [mtex-toolbox/mtex#1317] [Ask Anything]
Hello everyone ! 

I need to extract from EBSD data the crystal frame in order to make crystal plasticity simulations. As input, my finite element software need the x and y vectors of the crystal frame. So I have to know exactly how is the reference frame of the crystal in order to match it with the one in my finite element model frame.

I'm using   this command : 
ebsd  = EBSD.load(file,CS,'interface','ang','convertSpatial2EulerReferenceFrame','setting 2');
I looked at the documentation : https://mtex-toolbox.github.io/EBSDReferenceFrame.html. 
But in my simulated residual topographies, in comparison with experimental residual topographies (AFM), there is a mirror effect (perpendicular to A2).
To be sure off my well understanding, the euler angles given in the (A1 A2 A3) basis represent the documentation above in my case ?

Here is my code for the extraction of the crystals frames from selected grains  :
[Mtex2Ansys .txt](https://github.com/mtex-toolbox/mtex/files/7897232/Mtex2Ansys.txt)
Here is my EBSD map with the 3 frames : 
![EBSD](https://user-images.githubusercontent.com/95279492/150139316-120d2d85-d077-4661-9953-3fb00dd3b954.png)
(A1,A2,A3) => Euler reference frame
(xm,ym)     => map reference frame 
(X,Y,Z)       =>Software reference frame 
After a rotation of the map, I match the two frames : (X = A2 , Y = A3 , Z = A1)   

Thank you in advance for your help ! 

BOURCERET Alexandre

**Comment by kilir** [comment]
Hi,
if I understand correctly, yes. In mtex, the crystal coordinate system is aligned to X Y Z, the latter being identical to the the map/specimen reference frame.
Does that answer your question?
Cheers,
Rüdiger

**Reply by A-Bourceret** [reply]
Hi Rüdiger,
Thank you for your quick response !
Yes that 's my question, the import command above with the option 'convertSpatial2EulerReferenceFrame' with the 'setting 2' keeps everything in (A1,A2,A3) crystal reference frame ?
 And regarding the map, the Euler reference frame (A1,A2,A3) is oriented as I did in the pictures above ? (you have to click on the pictures to see those frames). 
 I suspect mirror effect because of the option of the plot that I use :
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','IntoPlane');
Without those option the images is not oriented like above. MTEXpref concern only plotting conventions ?
It means that I have to make rotations on the map and the Euler angles to have the correct orientation ? 

Thank you in advance for your help !

BOURCERET Alexandre

**Reply by kilir** [reply]
Hi,

> the import command above with the option 'convertSpatial2EulerReferenceFrame' with the 'setting 2' keeps everything in (A1,A2,A3) crystal reference frame ?

No, that is just one of the two ways to correct for the misalignment of coordinate frames used by manufacturers. Either change the Euler angles and keep x,y coordinates ('convertEuler2SpatialReferenceFrame')  or change the x,y coordinates and keep the Euler angles ('convertSpatial2EulerReferenceFrame').

>  And regarding the map, the Euler reference frame (A1,A2,A3) is oriented as I did in the pictures above ? (you have to click on the pictures to see those frames).
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','IntoPlane');


The appearance of XYZ on the screen is controlled by the plotting conventions, so in your case you choose to have X point to the east, Z into the plane, hence Y goes down. But those do not chance anything with respect to Euler angles or XY coordinates, simply how they are oriented on your screen.

> MTEXpref concern only plotting conventions ?

Yes

 > It means that I have to make rotations on the map and the Euler angles to have the correct orientation ?

Not sure if I understand correctly, but most likely not, unless the import options e.g. 'convertSpatial2EulerReferenceFrame' are not correct in your case. They are simply preset rotations which apply in most cases to Oxford systems and to Oim systems given the setting is right. There's a fair chance that at a different system setup, data origin or whatever, other rotations might be needed.

Cheers,
Rüdiger

**Reply by A-Bourceret** [reply]
Hi,

Thank you for your response and sorry for the late mine.
These precisions help me.

If I understand well, this orientation setting for the cartography depends on the systems used for extracting the data. 
I choose the option 'convertSpatial2EulerReferenceFrame' with 'setting 2' by comparison with a cartography given by the machine used for the experimental test. And I find the same map after that !

Thank you for your help again

Alexandre
